packages/trader/src/AppV2/Components/TradeParameters/RiskManagement/take-profit-and-stop-loss-input.tsx

```diff
@@ -0,0 +1,328 @@
+            clearTimeout(focus_timeout.current);
+        };
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [is_enabled, new_input_value]);
+
+    // TODO: remove all regex replacement, when comma will be supported by quill component with type number
+    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        is_be_response_received_ref.current = false;
+        let value = String(e.target.value)
+            .replace(',', '.')
+            .replace(/[.](?!\d*$)/g, '');
+        if (value.startsWith('.')) {
+            value = value.replace(/^\./, '0.');
+        }
+        if (value.length > 1) {
+            value = /^[0-]+$/.test(value) ? '0' : value.replace(/^0*/, '').replace(/^\./, '0.');
+        }
+
+        setNewInputValue(value);
+        updateParentRef({ field_name: type, new_value: value });
+    };
+
+    const onSave = () => {
+        // Prevent from saving if user clicks before BE validation
+        if (!is_be_response_received_ref.current && is_enabled) return;
+        WS.forget(subscription_id_ref.current);
+
+        if (error_text && is_enabled) return;
+        if (new_input_value === '' && is_enabled) {
+            setErrorText(
+                is_take_profit_input
+                    ? localize('Please enter a take profit amount.')
+                    : localize('Please enter a stop loss amount.')
+            );
+            return;
+        }
+
+        const is_tp_enabled = error_text ? false : is_enabled;
+        onChangeMultiple({
+            ...(is_take_profit_input ? { has_take_profit: is_tp_enabled } : { has_stop_loss: is_tp_enabled }),
+            ...(is_take_profit_input
+                ? {
+                      take_profit: error_text || new_input_value === '0' ? '' : new_input_value,
+                  }
+                : {
+                      stop_loss: error_text || new_input_value === '0' ? '' : new_input_value,
+                  }),
+            ...(is_tp_enabled ? { has_cancellation: false } : {}),
+        });
+        onActionSheetClose();
+    };
+
+    React.useEffect(() => {
+        setErrorText(initial_error_text ?? '');
+        updateParentRef({
+            field_name: is_take_profit_input ? 'tp_error_text' : 'sl_error_text',
+            new_value: initial_error_text ?? '',
+        });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [initial_error_text]);
+
+    React.useEffect(() => {
+        setInfo(info =>
+            (info.min_value !== min_value && min_value) || (info.max_value !== max_value && max_value)
+                ? { min_value, max_value }
+                : info
+        );
+    }, [min_value, max_value]);
+
+    return (
+        <React.Fragment>
+            <Component className={clsx('take-profit__wrapper', classname)}>
+                <div className='take-profit__content'>
+                    <Text>
+                        {is_take_profit_input ? (
```

Comment:
There's a lot of conditions like this across these files. The issue is imho that you treated both tp and sl as a single component. It would probably be way simpler to treat it as 2 simple separate components. In the future, feel free to simplify the complex components into few simple atomic components and it will simplify the development as well.